### ISSUE TITLE
Feature: create and update events directly from flows

### ISSUE THREAD

**Issue by jmdraper** [runely/calendar-homey#442]
My use case:
I rent my driveway out as a parking space using an app (Just Park). I have set up a system where the emails I receive from Just Park to notify me of a new booking are automatically parsed and they generate a web hook call to my homey passing the details of the booking including start and end date and times. These bookings can come in at any time and there may be several future bookings at any one time. I would like to trigger flows in my homey before / at the start and end times of the bookings.  

Suggested feature:
I would love to have the ability to create and delete (and maybe update, but could just do that via delete and then create) events directly from homey flows. I don't mean that these events should actually be added to a calendar since I understand that the subscription to ICS files in one-way. Instead I mean that I would like to have a new THEN card in the IcalCalendar app which creates a new event with specified start and end times and dates and a name/ID. This would then act exactly as if a new event had been ingested from an ICS calendar feed and then all the other iCalCalendar WHEN and AND cards would work as normal.

So in summary, two new THEN cards:
* create a new event called NAME with start date DATE and start time TIME, end date DATE and end time TIME
* delete an event named NAME

I don't think there is any other way to do this in Homey from within flows. The closest other thing I could find is the Action Scheduler app but that requires creation in advance of specific schedule devices which can then be manipulated with flows - whereas for my use case I need to create arbitrary number of new events from within flows.

Thank you for your consideration of this. I am a happy long-time user of this app and I want to thank you for actively maintaining this app which is a core part of my automation setup.

**Comment by runely** [comment]
That is quite an intriguing ide. I'll see what i can do 👍

**Comment by runely** [comment]
I have added Action flow cards for creating and deleting local events. I have added some decription here: https://github.com/runely/calendar-homey#actions

I've released this in a test version for now. Could you install this test version, try it out and give some feedback?
https://homey.app/no-no/app/no.runely.calendar/IcalCalendar/test/

**Comment by jmdraper** [comment]
Nice Easter project! Thank you - testing now and will let you know how I get on

**Comment by jmdraper** [comment]
I'm not sure whether I'm doing something wrong with the timestamp format or something but nothing seems to happen when I create a local event. When I manually ran the create local event card at the top of this flow it appeared to complete successfully but then none of the other triggers fired when the event was created / started / ended.

<img width="1049" alt="Screenshot 2023-04-07 at 13 17 19" src="https://user-images.githubusercontent.com/20240325/230607432-b189903e-d447-46c2-923f-834efbd0e7d2.png">

**Comment by jmdraper** [comment]
Never mind - just realised the timezone setting was the issue. Used the 'specific event is ongoing' card to inspect events and found it had created correctly but an hour later. Tested and working now with the apply timezone setting set to No - now it creates events at the times entered.

One other thing that would be useful would be the ability to add a description to the event as well.

This is looking really promising - thanks again!

**Comment by runely** [comment]
Yeah, that is described in the action card hint section. It's quite visible in a normal flow, but it's hidden behind the (`i`) in an advanved flow...

![image](https://user-images.githubusercontent.com/25814673/230654029-5a729f3d-57a0-4913-8957-134f08157524.png)

**Comment by runely** [comment]
New test version released which includes `description`: https://homey.app/no-no/app/no.runely.calendar/IcalCalendar/test/

**Comment by jmdraper** [comment]
Fantastic - thank you for this. Tested and working well.

**Comment by runely** [comment]
New test version released. In this one i added that `Event added` will trigger when a new local event is added.

https://homey.app/no-no/app/no.runely.calendar/IcalCalendar/test/

This version will be released when Athom has reviewed it. So i'm going to close this issue for now.

Feel free to comment here if anything seems off
